pwiz/utility/misc/SHA1.cpp

```diff
@@ -145,109 +145,68 @@ void CSHA1::Update(const UINT_8* pbData, UINT_32 uLen)
 }
 
 #ifdef SHA1_UTILITY_FUNCTIONS
-// Hash in file contents
 bool CSHA1::HashFile(const TCHAR* tszFileName)
 {
 	if(tszFileName == NULL) return false;
 
-	// first try to use memory-mapped I/O to maximize speed; on 32-bit systems this may fail for large files
-	try
-	{
-		const INT_64 fileSize = (INT_64)boost::filesystem::file_size(tszFileName);
-
-		// don't use a 32-bit process's entire address space, but 64-bit processes can just use the file size
-		const INT_64 maxRegionSize = sizeof(void*) == 8 ? fileSize : (1 << 30);
-		const INT_64 lMaxBuf = SHA1_MAX_FILE_BUFFER;
+	FILE* fpIn = _tfopen(tszFileName, _T("rb"));
+	if(fpIn == NULL) return false;
 
-		using namespace boost::interprocess;
-		file_mapping mmFile(tszFileName, read_only);

```

Comment:
There's a reason it's using memory mapped I/O: it's much faster! Or at least, it is much faster than the 2008 CSHA1 code. If you can show that the larger buffers in the newer CSHA1 code bring the performance up to par then we can switch back.

Reply:
The reason I had moved to the newer version was that the previous version loaded the entire file into memory, and some of my files for the fast scanning LIT in profile mode demultiplexed to 2x the number of scans were >50Gb. This provided a solution that could handle much larger input file sizes. I'm happy using whatever solution can scale to these larger file sizes.

I'm remembering now that I had wanted to merge the customized pwiz changes with the newer version, but didn't do that when I was just doing prototyping early on. I suppose I could try merging the pwiz customizations with this newer version and report back.

Reply:
Hmm. Memory mapping should load the whole file into virtual memory, but not physical memory. But after testing I've noticed it will use up all free physical memory as it churns through the file. It doesn't release the memory that it's already hashed. That is fixable. But I still benchmarked the several approaches. I think the upgrade to VS 2017 may have brought the two approaches closer together. Here's the performance of the various approaches on a 100GB file:
![image](https://user-images.githubusercontent.com/5011937/49316348-adb3b780-f4be-11e8-9843-667b10a08204.png)

It originally looks like the 1GB region does best, but when I tried it with the larger buffer size from the new code, it was slower by a couple minutes. So maybe shrink_by with the larger buffer is best. But not by a large margin and it uses considerably more memory (but 200MB isn't much when dealing with a 100GB file). I'll have to think about this some more.

Reply:
Ahh okay, would it be fair then to branch fixing the physical memory consumption off into a separate issue? I can revert my edits to the SHA1 files.

Reply:
Yes let's move it to a separate issue.

Reply:
Moved to https://github.com/ProteoWizard/pwiz/issues/352

Reply:
And reverted the changes in the latest commit